### ISSUE TITLE
Search box UI issues after geocoding

### ISSUE THREAD

**Issue by birbilis** [Viglino/ol-ext#561]
There are some UI issues with Search box control and reverse geocoding. Trying with SearchNominatim.

1) After reverse geocoding search, if one picks a location the searchbox control closes and they don't get to see the title of the location they have picked till they open search box again to see it as top entry in history. Should have an AutocloseAfterReverseGeocoding defaulting to false I think. An AutocloseAfterGeocoding defaulting to true would also be useful

2) If they had searched for some address before, then the searchbox will show those results everytime they reopen the searchbox, even if they had done reverse geocoding search (see steps described for issue 1). I believe reverse geocoding search should clear the search box. There seems to be a commented out "this.setInput('', true);" at SearchNominatime line 112:
https://github.com/Viglino/ol-ext/blob/4244a49b661f9567761abdd447365f962a512275/src/control/SearchNominatim.js#L112

not sure if it should be done before this._handleSelect(resp, true); though - ideally the reverse result should also show in the input, not just in the history so that you can edit/finetune the address and search forward this time

**Comment by birbilis** [comment]
btw, could have option after autocompleting the inputbox from reverse geocoding to do an automatic forward search to suggest other similar addresses. But I'd prefer the default to be to not do it. That is to just complete the inputbox with the found address to edit it and show the search history at the dropdown. User can then just press enter to search for that address

**Comment by Viglino** [comment]
I'll try to see what can be done...

**Comment by birbilis** [comment]
btw, it would also be nice to have an option to have the searchbox open from the start (apart from the one that would control whether it autocloses). Don't remember if it had such

**Comment by Viglino** [comment]
Hello
The search box open up filled with the response after reverse geocoding to lets you change it.
I've also added a `collapsed` option to open the search box on start and a `collapse()` method to show/hide the search box programmatically.

**Comment by birbilis** [comment]
Is this released?
I was using https://cdn.jsdelivr.net/gh/Viglino/ol-ext@3.1.17/dist/ol-ext.min.js but I don't see a 3.1.18 available there yet, not sure how often they grab the new versions if that is done automatically

btw, I noticed that 3.17 says 3.16 in its comments, 3.16 says 3.15 and so one, but 3.12 says 3.12 ok in its header whereas 3.13 also says 3.12. Not sure if it is that they grab the wrong versions or that you had forgotten to change the version in the header in some releases (assuming that is not an autogenerated thing from jsdelivr.net)

**Comment by birbilis** [comment]
Tried it at the examples page with Nominatim and seems to work fine

**Comment by Viglino** [comment]
Not yet released. I'll try to publish asap...

**Comment by Viglino** [comment]
3.1.18 is on the rail

**Comment by birbilis** [comment]
yep,
https://cdn.jsdelivr.net/gh/Viglino/ol-ext@3.1.18/dist/ol-ext.min.js
is live now
